### DISCUSSION TITLE
Anyone got a Pi 400 running?

### DISCUSSION THREAD

**Discussion by kapone3047** [Berny23/LD-ToyPad-Emulator#29] [Q&A]
Has anyone had luck getting this running on a Pi 400? It's all I have given Pi Zero's are impossible to purchase at the moment, and it is supposed to support USB gadget mode.

I've been able to install the toypad emulator (after a couple of hiccups) and have made sure dwc2 is enabled in config.txt and cmdline.txt for gadget mode. But when I connect the Pi to the PS5 nothing is detected.

I've tried using USB-C to USB-C (the PS5 provides 5A, more than enough), and a USB-A (from the Pi) to USB-C on the PS5, with no luck.

**Comment by cort1237** [comment]
I don't have a Pi4 so I can only give so much help, but I do know that it's the USB C port on the Pi you will have to use. That's the one that supports OTG connections.

It could be something wrong with the cables your using. USB C can be pretty difficult for knowing exactly what a cable is capable of. I'd fish around online for other Pi4 USB Gadgets and see what brand cables they recommend.

**Reply by kapone3047** [reply]
Thanks. I'll try some other cables around the house and see what suggestions I can find online.

**Comment by aussiehuddo** [comment]
Did you get this working? I am looking for a more easy to use guide on getting this up and running on a Pi4. 
I have installed Pi OS Lite (64-bit) and added wpa_supplicant.conf to /boot and the parameters to config.txt and cmdline.txt but can not see the device when scanning for IP address

EDIT: got it working following this guide up to step 5 https://desertbot.io/blog/headless-raspberry-pi-4-ssh-wifi-setup
1. install Pi OS lite 32-bit
2. enable ssh
3. add wifi network with wpa_supplicant.conf
4. plug in and scan for IP then SSH in and run the commands as listed on https://github.com/Berny23/LD-ToyPad-Emulator

Tested working on PS4 and PS5 with USB-C (from Pi) to USB A (on console)

**Comment by kapone3047** [comment]
Couldn't get it working on a Pi 400, finally got my hands on a Zero instead.

As for Pi 4, my understanding is that it's not capable of gadget mode so
won't work.

**Reply by aussiehuddo** [reply]
As you can see from my edit in above comment it does work on Pi 4 and it is capable of gadget mode

**Comment by kapone3047** [comment]
Sorry, only saw your post through email, not on the website so couldn't see
that.

Good to know it is possible
